test/test_web_api.py

```diff
@@ -0,0 +1,83 @@
+import pytest
+import sys
+from imp import reload
+import yaml
+import json
+
+from adr import query
+from app.app import app
+from urllib.parse import urlencode
+from adr.recipe import get_recipe_contexts
+
+from io import BytesIO, StringIO
+if sys.version_info > (3, 0):
+    IO = StringIO
+else:
+    IO = BytesIO
+
+
+class new_run_query(object):
```

Comment:
This could be fixture, like this one:
https://searchfox.org/mozilla-central/source/tools/tryselect/test/conftest.py#15